autonomi/src/client/data_types/pointer.rs

```diff
@@ -71,60 +109,167 @@ impl Client {
+                GetRecordError::RecordNotFound,
+            ))) => None,
+            Err(PointerError::Network(NetworkError::GetRecordError(
+                GetRecordError::SplitRecord { result_map },
+            ))) => result_map
+                .values()
+                .filter_map(|(record, _)| try_deserialize_record::<Pointer>(record).ok())
+                .max_by_key(|pointer: &Pointer| pointer.counter()),
+            Err(err) => {
+                return Err(err);
             }
         };
 
-        let payees = proof.payees();
+        let pointer = if let Some(p) = current {
+            let version = p.counter() + 1;
+            Pointer::new(owner, version, target)
+        } else {
+            warn!("Pointer at address {address:?} cannot be updated as it does not exist, please create it first or wait for it to be created");
+            return Err(PointerError::CannotUpdateNewPointer);
+        };
 
+        // prepare the record to be stored
         let record = Record {
             key: NetworkAddress::from_pointer_address(address).to_record_key(),
-            value: try_serialize_record(
-                &(proof, &pointer),
-                RecordKind::DataWithPayment(DataTypes::Pointer),
-            )
-            .map_err(|_| PointerError::Serialization)?
-            .to_vec(),
+            value: try_serialize_record(&pointer, RecordKind::DataOnly(DataTypes::Pointer))
+                .map_err(|_| PointerError::Serialization)?
+                .to_vec(),
             publisher: None,
             expires: None,
         };
-
         let get_cfg = GetRecordCfg {
             get_quorum: Quorum::Majority,
             retry_strategy: Some(RetryStrategy::default()),
             target_record: None,
             expected_holders: Default::default(),
         };
-
         let put_cfg = PutRecordCfg {
             put_quorum: Quorum::All,
             retry_strategy: None,
             verification: Some((VerificationKind::Crdt, get_cfg)),
-            use_put_record_to: Some(payees),
+            use_put_record_to: None,
         };
 
         // store the pointer on the network
-        debug!("Storing pointer at address {address:?} to the network");
+        debug!("Updating pointer at address {address:?} to the network");
         self.network
             .put_record(record, &put_cfg)
             .await
             .inspect_err(|err| {
-                error!("Failed to put record - pointer {address:?} to the network: {err}")
+                error!("Failed to update pointer at address {address:?} to the network: {err}")
             })?;
 
-        Ok(address)
+        Ok(())
     }
 
     /// Calculate the cost of storing a pointer
-    pub async fn pointer_cost(&self, key: SecretKey) -> Result<AttoTokens, PointerError> {
-        let pk = key.public_key();
-        trace!("Getting cost for pointer of {pk:?}");
+    pub async fn pointer_cost(&self, key: PublicKey) -> Result<AttoTokens, PointerError> {
```

Comment:
not seeing this to be used by the pointer_put ?

Reply:
Every type has cost estimation function. This estimate is not used by the put no.

Reply:
sry, I mean the pointer_put carries the `cost quoting` work as well, which actually can just call this `pointer_cost` function to reduce the duplication, and avoid future code update glitch (one updated, the other not)

combined with the [point_put to be private](https://github.com/maidsafe/autonomi/pull/2658/files#r1926834712) comment, I think the pointer API shall be:
```rust
pub fn pointer_create() { creations; pointer_cost(); pointer_put(with_payment); }
pub fn pointer_update() { fetch; update; pointer_put(without_payment); }
pub fn pointer_cost() { quoting only; }
fn pointer_put() { if with_payment { make_payment; put PointerWithPayment } else { put Pointer } }
```